File: proxy/tests/script_engine/stop_invoked.py
import json
from roxy import Extension


class StopInvoked(Extension):
    def start(self):
        self.count = 10
        self.out_file = ""

    def request(self, flow):
        flow.request.body.text = str(self.count)
        self.count += 1

    def response(self, flow):
        self.out_file = flow.response.body.text
        flow.response.body.text = str(self.count)
        self.count += 1

    def stop(self):
        with open(self.out_file, "w") as f:
            json.dump({"stopped": True, "count": self.count}, f)


Extensions = [StopInvoked()]

File: proxy/tests/script_engine/stop_invoked.lua
pcall(require, "../../script_libs/lua/roxy.lua")
local out_file = ""
local count = 0
---@type Extension
local stop_invoked = {
	start = function()
		count = 10
	end,
	request = function(flow)
		flow.request.body.text = "" .. count
		count = count + 1
	end,
	response = function(flow)
		out_file = flow.response.body.text
		flow.response.body.text = "" .. count
	end,
	stop = function()
		local f = assert(io.open(out_file, "w"))
		f:write(string.format('{"stopped":true,"count":%d}', count))
		f:close()
	end,
}
Extensions = { stop_invoked }
